src/modules/Builder/pages/BuilderApplet/BuilderApplet.tsx

```diff
@@ -82,18 +89,20 @@ export const BuilderApplet = () => {
         }
       })();
     }
-  }, [loadingStatus, isNewApplet]);
+  }, [loadingStatus, isNewApplet, appletResponseType]);
 
   useEffect(() => {
-    const isFromLibrary = storage.getItem(LocalStorageKeys.IsFromLibrary) as boolean;
-    if (isFromLibrary) {
-      setIsFromLibrary(true);
+    if (isFromLibrary && isNewApplet && dataFromLibrary) {
```

Comment:
pls use a variable for bunch of conditions and use early return to avoid nesting

Reply:
fixed